### PR TITLE
[10.x] Add `content` method to Vite

### PR DESCRIPTION
**Pull request by michael-rubel**
## About
This PR adds a method to retrieve compiled content for the given file available in the Vite manifest. It allows us to pack our Vite assets inline when needed.

## Example use case
You want to create an HTML page that needs to be open or sent as a single document (for example, as a mail attachment), and to work offline (without access to the internet) so the asset should be injected into the document for the browser to work on the local file without firing the HTTP requests.

## Usage example
```php
<style>{!! Vite::content('resources/scss/file.scss') !!}</style>
```

### CODE REVIEW

File: src/Illuminate/Foundation/Vite.php [hunk 1]
```diff
@@ -655,6 +655,24 @@ public function asset($asset, $buildDirectory = null)
...
+     */
+    public function content($asset, $buildDirectory = null)
+    {
+        $buildDirectory ??= $this->buildDirectory;
+
+        $chunk = $this->chunk($this->manifest($buildDirectory), $asset);
+
+        $path = public_path($buildDirectory.'/'.$chunk['file']);
+
+        return file_get_contents($path);
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Vite.php:673]: I don't think we should do an unchecked call to file_get_contents.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Foundation/Vite.php (hunk 1 above)
Comment (michael-rubel) [reply src/Illuminate/Foundation/Vite.php:677]: @GrahamCampbell added the check. Will it be satisfactory for most of the cases?

File: src/Illuminate/Foundation/Vite.php [hunk 2]
```diff
@@ -655,6 +655,28 @@ public function asset($asset, $buildDirectory = null)
...
+    public function content($asset, $buildDirectory = null)
+    {
+        $buildDirectory ??= $this->buildDirectory;
+
+        $chunk = $this->chunk($this->manifest($buildDirectory), $asset);
+
+        $path = public_path($buildDirectory.'/'.$chunk['file']);
+
+        if (! is_file($path) || ! file_exists($path)) {
+            return '';
```
Comment (jessarcher) [inline src/Illuminate/Foundation/Vite.php:674]: I wonder whether this should throw an exception, similar to what the `chunk` method will do if the file doesn't exist in the manifest.

**Comment by jessarcher** [comment]
This sounds pretty useful, and I've run into similar scenarios myself.

My only concern is that the local dev behaviour might be unexpected or confusing when running the Vite dev server. It will fail if you haven't run a build since adding the asset. Otherwise, it will return the contents from the last build, ignoring any changes you've made to the source file. You'd need to run Vite in "watch" mode instead of HMR mode if you wanted to automatically have the latest changes while developing.

File: src/Illuminate/Foundation/Vite.php (hunk 2 above)
Comment (michael-rubel) [reply src/Illuminate/Foundation/Vite.php]: @jessarcher I've added the exception.
Can you have a look if that's what you'd expect?

File: src/Illuminate/Foundation/Vite.php [hunk 3]
```diff
@@ -655,6 +655,30 @@ public function asset($asset, $buildDirectory = null)
...
+    public function content($asset, $buildDirectory = null)
+    {
+        $buildDirectory ??= $this->buildDirectory;
+
+        $chunk = $this->chunk($this->manifest($buildDirectory), $asset);
+
+        $path = public_path($buildDirectory.'/'.$chunk['file']);
+
+        if (! is_file($path) || ! file_exists($path)) {
+            throw new Exception("Unable to locate file at path: {$path}.");
```
Comment (jessarcher) [inline src/Illuminate/Foundation/Vite.php:676]: I wonder if including "Vite" might help.

E.g.

```suggestion
            throw new Exception("Unable to locate file from Vite manifest: {$path}.");
```

Hopefully, it's distinct enough from the error when the file doesn't exist in the manifest:

```php
            throw new Exception("Unable to locate file in Vite manifest: {$file}.");
```

**Comment by jessarcher** [comment]
@michael-rubel You could try calling `file_get_contents` on the dev server URL for the asset, but I'm not sure how well that would work. For something like CSS, it might be okay, but for JavaScript, it might cause more issues and confusion than it solves :sweat_smile: 

It's probably fine as-is. Just a caveat to be aware of when using it.

**Pull request merged**